api/client/webclient/webclient.go

```diff
@@ -376,6 +374,38 @@ func tunnelAddr(proxyAddr string, settings ProxySettings) (string, error) {
 	return net.JoinHostPort(host, tunnelPort), nil
 }
 
+// tunnelAddrForTLSRouting returns reverse tunnel proxy address for proxy supporting TLS Routing.
+func tunnelAddrForTLSRouting(proxyAddr string, settings ProxySettings) (string, error) {
+	if settings.SSH.PublicAddr != "" {
+		// Check if PublicAddr contains a port number.
+		if _, err := extractPort(settings.SSH.PublicAddr); err == nil {
+			return extractHostPort(settings.SSH.PublicAddr)
+		}
+		// Get port number from proxyAddr or use default one.
+		port := strconv.Itoa(defaults.ProxyWebListenPort)
+		if webPort, err := extractPort(proxyAddr); err == nil {
+			port = webPort
+		}
+
+		if host, err := extractHost(settings.SSH.PublicAddr); err == nil {
+			return net.JoinHostPort(host, port), nil
+		}
+	}
+
+	// Got proxyAddr with a port number for instance: proxy.example.com:3080
+	if _, err := extractPort(proxyAddr); err == nil {
+		return proxyAddr, nil
+	}
+	host, err := extractHost(proxyAddr)
+	if err != nil {
+		return "", trace.Wrap(err, "failed to parse the given proxy address")
+	}
+
+	// Got proxy address without a port like:  proxy.example.com
+	// If proxyAddr doesn't contain any port it means that HTTPS port should be used.
+	return net.JoinHostPort(host, "443"), nil
```

Comment:
Why 443? Isn't default proxy port 3080?

Reply:
This code is invoked only if `Find web` API  call succeed. 


if `proxyAddr` format doesn't contain port number it means that the default `https` port was used in Find API call: 
```
	endpoint := fmt.Sprintf("https://%s/webapi/find", proxyAddr)
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, endpoint, nil)
```

Reply:
Hmm I see. This is a little confusing, let's at least expand the comment to explain. Also, do we have a constant for 443?

Reply:
Extended the comment.  
>  Also, do we have a constant for 443?
there is gravitational/teleport.StandardHTTPSPort const but importing `gravitational/teleport`  in teleport/api` pkg is not possible thus I move this const to api defaults.